packages/fluentui/react-northstar/src/utils/positioner/usePopper.ts

```diff
@@ -0,0 +1,391 @@
+
+        ...(typeof userModifiers === 'function' ? userModifiers(target, container, arrow) : userModifiers),
+
+        /**
+         * This modifier is necessary to retain behaviour from popper v1 where untethered poppers are allowed by
+         * default. i.e. popper is still rendered fully in the viewport even if anchor element is no longer in the
+         * viewport.
+         */
+        unstable_disableTether && {
+          name: 'preventOverflow',
+          options: { altAxis: unstable_disableTether === 'all', tether: false },
+        },
+
+        flipBoundary && {
+          name: 'flip',
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(container, flipBoundary),
+          },
+        },
+        overflowBoundary && {
+          name: 'preventOverflow',
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(container, overflowBoundary),
+          },
+        },
+
+        {
+          name: 'onUpdate',
+          enabled: true,
+          phase: 'afterWrite' as PopperJs.ModifierPhases,
+          fn: handleStateUpdate,
+        },
+
+        autoSize && {
+          // Similar code as popper-maxsize-modifier: https://github.com/atomiks/popper.js/blob/master/src/modifiers/maxSize.js
+          // popper-maxsize-modifier only calculates the max sizes. This modifier applies the max sizes when overflow is detected
+          name: 'applyMaxSize',
+          enabled: true,
+          phase: 'beforeWrite' as PopperJs.ModifierPhases,
+          requiresIfExists: ['offset', 'preventOverflow', 'flip'],
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(container, overflowBoundary),
+          },
+          fn({ state, options }: PopperJs.ModifierArguments<{}>) {
+            const overflow = PopperJs.detectOverflow(state, options);
+            const { x, y } = state.modifiersData.preventOverflow || { x: 0, y: 0 };
+            const { width, height } = state.rects.popper;
+            const [basePlacement] = state.placement.split('-');
+
+            const widthProp: keyof PopperJs.SideObject = basePlacement === 'left' ? 'left' : 'right';
+            const heightProp: keyof PopperJs.SideObject = basePlacement === 'top' ? 'top' : 'bottom';
+
+            if (overflow[widthProp] > 0 && (autoSize === true || autoSize === 'width')) {
+              state.styles.popper.maxWidth = `${width - overflow[widthProp] - x}px`;
+            }
+            if (overflow[heightProp] > 0 && (autoSize === true || autoSize === 'height')) {
+              state.styles.popper.maxHeight = `${height - overflow[heightProp] - y}px`;
+            }
+          },
+        },
+
+        /**
+         * This modifier is necessary in order to render the pointer. Refs are resolved in effects, so it can't be
+         * placed under computed modifiers. Deep merge is not required as this modifier has only these properties.
+         */
+        {
```

Comment:
This modifier has been moved down compared to `Popper`. Could you please explain why?

Reply:
In v2 order of modifiers [is determined](https://popper.js.org/docs/v2/modifiers/#phase) by `phase`s and `requires`/`requiresIfExists` options ([orderModifiers()](https://github.com/popperjs/popper-core/blob/e1c1a37f2b2d929262953996fb35e4a9a8c9a2ed/src/utils/orderModifiers.js#L47)).

I modified an order there to have more sense (_from my POV_), if you have another suggestions, please post them 😉